### ISSUE TITLE
Bug: __FILE__ expands to "?" in parameter lists of initialize methods

### ISSUE THREAD

**Issue by drosehn** [crystal-lang/crystal#4250]
This is on:
```cr
# crystal --version
Crystal 0.21.1 (2017-03-18) LLVM 4.0.0
# uname -a
Darwin my.host 14.5.0 Darwin Kernel Version 14.5.0: Sun Sep 25 22:07:15 PDT 2016; root:xnu-2782.50.9~1/RELEASE_X86_64 x86_64
# # aka macOS 10.10.5 (Yosemite)
```
This issue was mentioned by someone on google-groups as `Bug: __FILE__ expands to "?"`, but it looks like they never wrote up a github-issue about it.  The same issue can be seen with `__DIR__`, but I'll limit the sample program to `__FILE__`.  The program:
```cr
class FileClass
  def initialize(fname = "#{__FILE__}", f2 = __FILE__)
    printf "__FILE__:  %s\n", __FILE__
    puts "   fname:  #{fname}"
    puts "   vs f2:  #{f2}"
  end

  def other(fother = "#{__FILE__}", f3 = __FILE__)
    puts ""
    printf "__FILE__:  %s\n", __FILE__
    puts "  fother:  #{fother}"
    puts "   vs f3:  #{f3}"
  end
end

fc = FileClass.new
fc.other
```
produces output of:
```cr
# crystal run /tmp/PR_file.cr
__FILE__:  /tmp/PR_file.cr
   fname:  /tmp/PR_file.cr
   vs f2:  ?

__FILE__:  /tmp/PR_file.cr
  fother:  /tmp/PR_file.cr
   vs f3:  /tmp/PR_file.cr
```
I got the idea of `fname = "#{__FILE__}"` vs `f2 = __FILE__` because at some point in my testing the compiler complained that it didn't know the type of `__FILE__`, so I thought that `"#{__FILE__}"` would make it clear that the type was `String`.

**Comment by asterite** [comment]
Reduced:

```crystal
class Foo
  def initialize(x = __FILE__)
    p x
  end
end

Foo.new
```

The above incorrectly prints "?"

This is kind of hard to implement so it will have to wait.

**Comment by j8r** [comment]
Fixed on Crystal 0.25.0.
